filetablemodel.cpp

```diff
@@ -17,75 +17,66 @@ FileTableModel::FileTableModel(QObject *parent)
     // than with lists. But it is easier to fill table using lists as
     // of they could change their size easily. So we build vector of
     // lists and then convert it to vector of vectors.
-    // TODO: It seems to be not optimal structure. May pointers be faster?
-    //       How does it work being assigned? Does it make full copy?
-    // TODO: Is string an appropriate type for fields? May bytes be
-    //       better?
+
     QVector<QStringList> data(columnCount());
-    // We read and parse the file.
-    // We create and fill our internal model representation.
-    // TODO: Restore could call us with empty name. We crash.
-    QFile file(fileName);
-    if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
-        return false;
-    while (!file.atEnd()) {
-        QString line = file.readLine();
-        // TODO: right? Should not we keep \r in middle of line?
-        line.remove(QRegExp("\\r?\\n"));
-        // TODO: Parse gecos well.
-        // TODO: Make customizable separator. It should be an option.
-        QStringList fields = line.split(':');
-        // TODO: Is it safe to show untrusted input in gui?
-        int column = 0;
-        // NOTE: When we want we change lists we use [] as of .at()
-        //       gives us only const.
-        if (fields.size() == 1) {
-            // Lonely hash
-            // TODO: Special mark to show that ? is not from file. Color? How?
-            data[column++].append("?");
-            data[column++].append("");
-            data[column++].append(fields.at(0));
-        } else if (fields.size() >= 3 && fields.at(2).indexOf(QRegExp("^[0-9a-fA-F]{32}$")) == 0) {
-            // Pwdump format
-            data[column++].append(fields.at(0));
-            data[column++].append("");
-            data[column++].append(fields.at(2));
-            // TODO: It is not good to pack it so. Parse gecos well.
-            // TODO: split and join back seem slower than optimal.
-            fields.removeAt(2);
-            fields.removeAt(0);
-            data[column++].append(fields.join(":"));
-        } else {
-            // user:hash:other
-            data[column++].append(fields.at(0));
-            data[column++].append("");
-            data[column++].append(fields.at(1));
-            fields.removeAt(1);
-            fields.removeAt(0);
-            data[column++].append(fields.join(":"));
+    for(int fileCount = 0; fileCount < fileNames.size(); fileCount++)
+    {
+        // We read and parse the file.
+        // We create and fill our internal model representation.
+        QFile file(fileNames[fileCount]);
+        if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
+            return false;
+        while (!file.atEnd()) {
+            QString line = file.readLine();
+            // TODO: right? Should not we keep \r in middle of line?
+            line.remove(QRegExp("\\r?\\n"));
+            // TODO: Parse gecos well.
```

Comment:
On Fri, May 22, 2015 at 11:29:28AM -0700, math07 wrote:
> > -            data[column++].append(fields.at(1));
> > -            fields.removeAt(1);
> > -            fields.removeAt(0);
> > -            data[column++].append(fields.join(":"));
> > +    for(int fileCount = 0; fileCount < fileNames.size(); fileCount++)
> > +    {
> > +        // We read and parse the file.
> > +        // We create and fill our internal model representation.
> > +        QFile file(fileNames[fileCount]);
> > +        if (!file.open(QIODevice::ReadOnly | QIODevice::Text))
> > +            return false;
> > +        while (!file.atEnd()) {
> > +            QString line = file.readLine();
> > +            // TODO: right? Should not we keep \r in middle of line?
> > +            line.remove(QRegExp("\\r?\\n"));
> > +            // TODO: Parse gecos well.
> 
> @AlekseyCherepanov What do you mean by parse gecos well ? Is there a problem with current gecos ?

Actually there are several fields: uid, gid, gecos, home, shell.
Currently all of them are in 1 field called gecos. Maybe we should use
--show=types to do the parsing, maybe no because there is core without
--show-types